### DISCUSSION TITLE
Why can't we change Users schema in GUI?

### DISCUSSION THREAD

**Discussion by artemdzyuba** [supabase/supabase#3142] [Questions] [answered]
For some reason, we can't add or delete a column to the Users table in the editor - unlike to any other table. But we can alter Users via a querry. So why was that artificially restricted?

**Comment by steve-chavez** [comment, marked as answer]
Supabase Auth component([gotrue](https://github.com/supabase/gotrue/)) depends on the `auth.users` table and runs migrations against it. So really `auth.users` is a private table and you shouldn't change it - doing so might cause gotrue to fail.

> But we can alter Users via a querry.

This won't be possible in the future, I don't recommend relying on it.

Instead you can create another table, like `users_details`, and add it a foreign key to `auth.users`, here you can add all the columns you need.

**Reply by artemdzyuba** [reply]
Yes, now I got it, thank you.

**Reply by artemdzyuba** [reply]
@steve-chavez ,  when I try to add a Foreign key to a new table, I can't address the auth.users table. Does it mean that I can;t explicetly define this reference as a Foreign key? I can, of course, just create an UUID field without putting it as a foreing, but this looks strange a bit.

**Reply by MildTomato** [reply]
the UI currently doesn't support (not yet..) adding foreign keys to non public schema.

this might be helpful though, there's a cheat sheet for setting up _user details_ table
 https://supabase-cheatsheet.vercel.app/docs/auth/public-profiles-table
 
 and official docs on this
 https://supabase.io/docs/guides/auth/managing-user-data
 
 you don't have to set up a trigger though, that's optional
